Erreur BotEnTrain 403 - Arrêt du plan gratuit pour certaines API Twitter - Arrêt du service

June 15 2023

Failed - Twitter API Error 403 

Log extract (some value replaced by 'xxx')
`````
Jun 15 09:05:06 AM  [2023-06-15T09:05:06.881] [INFO] FilmBTP - config {"doSimulate":false}
Jun 15 09:05:07 AM  [2023-06-15T09:05:07.231] [ERROR] FilmBTP - ApiResponseError: Request failed with code 403
Jun 15 09:05:07 AM      at RequestHandlerHelper.createResponseError (/opt/render/project/src/node_modules/twitter-api-v2/dist/cjs/client-mixins/request-handler.helper.js:103:16)
Jun 15 09:05:07 AM      at RequestHandlerHelper.onResponseEndHandler (/opt/render/project/src/node_modules/twitter-api-v2/dist/cjs/client-mixins/request-handler.helper.js:252:25)
Jun 15 09:05:07 AM      at Gunzip.emit (events.js:376:20)
Jun 15 09:05:07 AM      at endReadableNT (internal/streams/readable.js:1336:12)
Jun 15 09:05:07 AM      at processTicksAndRejections (internal/process/task_queues.js:82:21) {
Jun 15 09:05:07 AM    error: true,
Jun 15 09:05:07 AM    type: 'response',
Jun 15 09:05:07 AM    code: 403,
Jun 15 09:05:07 AM    headers: {
Jun 15 09:05:07 AM      date: 'Thu, 15 Jun 2023 07:05:07 UTC',
Jun 15 09:05:07 AM      perf: '7626143928',
Jun 15 09:05:07 AM      server: 'tsa_f',
Jun 15 09:05:07 AM      'set-cookie': [
Jun 15 09:05:07 AM        'guest_id=v1%xxx; Max-Age=34214400; Expires=Mon, 15 Jul 2024 07:05:07 GMT; Path=/; Domain=.twitter.com; Secure; SameSite=None'
Jun 15 09:05:07 AM      ],
Jun 15 09:05:07 AM      'api-version': '2.64',
Jun 15 09:05:07 AM      'content-type': 'application/json; charset=utf-8',
Jun 15 09:05:07 AM      'cache-control': 'no-cache, no-store, max-age=0',
Jun 15 09:05:07 AM      'content-length': '328',
Jun 15 09:05:07 AM      'x-access-level': 'read',
Jun 15 09:05:07 AM      'x-frame-options': 'SAMEORIGIN',
Jun 15 09:05:07 AM      'content-encoding': 'gzip',
Jun 15 09:05:07 AM      'x-transaction-id': 'xxx',
Jun 15 09:05:07 AM      'x-xss-protection': '0',
Jun 15 09:05:07 AM      'x-rate-limit-limit': '40000',
Jun 15 09:05:07 AM      'x-rate-limit-reset': '1686813607',
Jun 15 09:05:07 AM      'content-disposition': 'attachment; filename=json.json',
Jun 15 09:05:07 AM      'x-content-type-options': 'nosniff',
Jun 15 09:05:07 AM      'x-rate-limit-remaining': '39999',
Jun 15 09:05:07 AM      'strict-transport-security': 'max-age=631138519',
Jun 15 09:05:07 AM      'x-response-time': '108',
Jun 15 09:05:07 AM      'x-connection-hash': 'xxx',
Jun 15 09:05:07 AM      connection: 'close'
Jun 15 09:05:07 AM    },
Jun 15 09:05:07 AM    rateLimit: { limit: 40000, remaining: 39999, reset: 1686813607 },
Jun 15 09:05:07 AM    data: {
Jun 15 09:05:07 AM      client_id: 'xxx',
Jun 15 09:05:07 AM      detail: 'When authenticating requests to the Twitter API v2 endpoints, you must use keys and tokens from a Twitter developer App that is attached to a Project. You can create a project via the developer portal.',
Jun 15 09:05:07 AM      registration_url: 'https://developer.twitter.com/en/docs/projects/overview',
Jun 15 09:05:07 AM      title: 'Client Forbidden',
Jun 15 09:05:07 AM      required_enrollment: 'Appropriate Level of API Access',
Jun 15 09:05:07 AM      reason: 'client-not-enrolled',
Jun 15 09:05:07 AM      type: 'https://api.twitter.com/2/problems/client-forbidden'
Jun 15 09:05:07 AM    }
Jun 15 09:05:07 AM  }
`````

This issue is not only for BotEnTrain but touch many twitterAPI users
cf. https://twittercommunity.com/t/v2-suddenly-getting-client-not-enrolled-today/195456/31

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -39,22 +39,3 @@ jobs:
         npm install
         echo "show outdated (if any)"
         npm outdated --depth=3 || echo "you must think about update your dependencies :)"
-
-    - name: Run tests
-      run: |
-        echo "${{ secrets.TEST_ENV_FILE }}" >> ./env/.env.test
-        npm run ci-test
-
-    - name: Report coverage
-      continue-on-error: true
-      uses: romeovs/lcov-reporter-action@v0.2.16
-      with:
-        github-token: ${{ secrets.GITHUB_TOKEN }}
-        lcov-file: ./coverage/lcov.info
-
-    - name: Npm audit
-      uses: oke-py/npm-audit-action@v1.7.1
-      with:
-        audit_level: moderate
-        github_token: ${{ secrets.GITHUB_TOKEN }}
-        dedupe_issues: true

--- a/src/www/public/main.css
+++ b/src/www/public/main.css
@@ -27,3 +27,11 @@
 .illustrateTxt {
   float: right;
 }
+
+.msgContainer {
+  border: 2px solid yellow;
+  padding: 20px;
+}
+.msgTitle {
+  font-size: 20px;
+}

--- a/src/www/views/pages/index.ejs
+++ b/src/www/views/pages/index.ejs
@@ -25,6 +25,20 @@
 
 <body>
 <div>
+
+
+  <div class="msgContainer">
+    <div class="msgTitle">Arrêt du service</div>
+    <div class="msgContent">
+    Depuis le 13 juin, le plan d'accès gratuit à certaines API twitter a changé
+    (<a href="https://twittercommunity.com/t/deprecation-of-legacy-access-tiers-in-the-twitter-api/196162" target="_new">annonce</a>). <br/>
+    Il n'est désormais plus possible de faire des recherches de tweets sans prendre un plan payant.<br/>
+    Le service <code>@botentrain1</code> s'arrête donc jusque nouvel ordre. cf. <a target="_new" href="https://github.com/boly38/botEnTrain/issues/99">ticket #99</a>.
+    </div>
+  </div>
+
+
+
   <div class="panel-heading">
     <ul class="nav nav-tabs">
       <li id="logs-menu">

